mig/shared/transferfunctions.py

```diff
@@ -180,49 +229,61 @@ def modify_data_transfers(configuration, client_id, transfer_dict, action,
     elif action == "delete":
         del transfers[transfer_id]
     else:
-        return (False, "Invalid action %s on data transfers" % action)
+        if do_lock:
```

Comment:
More on the logic side, I think the addition of "one-off" locking as opposed to having a good abstraction for this is not ideal. We already have various objects that wrap file handling, so my take would is that if we must opt-in to this functionality via a `do_lock` flag (I believe this would be better named `with_lock`) it's effect should be to something like changing the file class we open via with - the result ending up as something akin to:

```
FileClass = LockedLocalFile if with_lock else LockFile

with FileClass(<file_path>) as my_file:
   ...
```

with all the unlocking abstracted from the caller here into the LockedLocalFile, something that would be throughly tested once and centrally thus we'd not need to have so many tests here and instead it would be sufficient to check that with_lock caused the locked file class to be selected.